### PR TITLE
Fix errcheck issues in server/channels/app/platform/service.go

### PR DESCRIPTION
**Pull request by AulakhHarsh**
#### Summary
Fix errcheck issues in server/channels/app/platform/service.go

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/29356
Jira https://mattermost.atlassian.net/browse/MM-61770

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/platform/service.go [hunk 1]
```diff
@@ -370,7 +370,10 @@ func New(sc ServiceConfig, options ...Option) (*PlatformService, error) {
 		}
 	})
 
-	ps.SearchEngine.UpdateConfig(ps.Config())
+	if err := ps.SearchEngine.UpdateConfig(ps.Config()); err != nil {
+		ps.logger.Error("Failed to update search engine config", mlog.Err(err))
```
Comment (hanzei) [inline server/channels/app/platform/service.go:374]: Nit pick:
```suggestion
		ps.Log().Warn("Failed to update search engine config", mlog.Err(err))
```

**Review by hanzei — APPROVED** [review]
Thanks @AulakhHarsh :+1:

**Comment by AulakhHarsh** [comment]
@marianunez requesting your review on this.

**Comment by AulakhHarsh** [comment]
> @marianunez requesting your review on this.

@marianunez bumping this up.

**Comment by marianunez** [comment]
> > @marianunez requesting your review on this.
> 
> @marianunez bumping this up.

@AulakhHarsh  Apologies for the delay as I was out for the holiday here in the US. I will review shortly.

**Comment by mvitale1989** [comment]
FYI the failing testcases are unrelated to this PR, and already fixed on master branch, so they would've succeeded after an `/update-branch`.
I'll override the E2E result to green to allow merging

**Pull request merged**